app/src/main/java/com/example/tasky/feature_agenda/presentation/AgendaViewState.kt

```diff
@@ -1,23 +1,55 @@
 package com.example.tasky.feature_agenda.presentation
 
 import com.example.tasky.common.domain.error.DataError
+import com.example.tasky.common.presentation.model.AgendaItemType
 import com.example.tasky.common.presentation.util.toLong
 import com.example.tasky.feature_agenda.presentation.model.CalendarDay
 import com.vanpra.composematerialdialogs.MaterialDialogState
 import java.time.LocalDate
 
 /* define viewState as UI elements that change throughout app usage */
 data class AgendaViewState(
-     val datePickerDialogState: MaterialDialogState = MaterialDialogState(),
-     val showLogoutDropdown: Boolean = false,
-     val showFabDropdown: Boolean = false,
-     val monthSelected: Int = LocalDate.now().monthValue,
-     val daySelected: Int = LocalDate.now().dayOfMonth,
-     val calendarDays: List<CalendarDay> = emptyList(),
-     val showLoadingSpinner: Boolean = false,
-     val showErrorDialog: Boolean = false,
-     val dataError: DataError = DataError.Network.UNKNOWN,
-     val headerDateText: String = "",
-     val yearSelected: Int = LocalDate.now().year,
-     val dateSelected: Long = LocalDate.now().toLong()
-)
\ No newline at end of file
+    val datePickerDialogState: MaterialDialogState = MaterialDialogState(),
+    val showLogoutDropdown: Boolean = false,
+    val showFabDropdown: Boolean = false,
+    val monthSelected: Int = LocalDate.now().monthValue,
+    val daySelected: Int = LocalDate.now().dayOfMonth,
+    val calendarDays: List<CalendarDay> = emptyList(),
+    val showLoadingSpinner: Boolean = false,
+    val showErrorDialog: Boolean = false,
+    val dataError: DataError = DataError.Network.UNKNOWN,
+    val headerDateText: String = "",
+    val yearSelected: Int = LocalDate.now().year,
+    val dateSelected: Long = LocalDate.now().toLong(),
+    val agendaItems: List<AgendaItem>? = null
+)
+
+sealed class AgendaItem(
+    open val title: String,
+    open val details: String,
+    open val date: Long,
+    open val isChecked: Boolean,
+    open val cardType: AgendaItemType
+) {
```

Comment:
You put this model here in presentation which is all fine if you just need it to display something. But agenda items and what data they consist of are such a core concept of this app, which is also needed in all layers. That's always a strong hint that something should be a domain model instead, as I also mentioned in an earlier comment. 

I know it's hard to wrap your head around these different layers and models. What helps me is to think of the app from a complete non-technical perspective. If there was no UI you had to worry about, no libraries, no APIs or databases, but just think of the problem the app solves (in other words, it's domain) - would there be a use for this model? 
This app solves the problem that you won't forget about your events and tasks anymore. That sentence already contains the information that it must be about some sort of event or task. So without including any technical details in the description of the app, we can't describe it without mentioning these core concepts of agenda items. Therefore, this must be a domain model.

Different example: If you had a social media app, like Instagram, what could be valid domain models? One could be a `Post` model for example, which includes information what a post consists of. A `Post` in the domain of Instagram is a completely different model than a `Post` would be in the context of Twitter, since those are two different domains. That's again a strong indication towards a domain model.